build/build.sh

```diff
@@ -306,19 +306,36 @@ function InstallDotNetCli {
 }
 
 function InstallRepoToolset {
+  CreateDirectory "$RepoToolsetDir"
   ReadJson "$RepoRoot/global.json" "RoslynTools.RepoToolset"
   RepoToolsetVersion=$readjsonvalue
-  RepoToolsetDir="$NuGetPackageRoot/roslyntools.repotoolset/$RepoToolsetVersion/tools"
-  RepoToolsetBuildProj="$RepoToolsetDir/Build.proj"
+  RepoToolsetLocationFile="$RepoToolsetDir/$RepoToolsetVersion.txt"
+
+  if [[ -a "$RepoToolsetLocationFile" ]]; then
+    local path=`cat $RepoToolsetLocationFile`
+    if [[ -a "$path" ]]; then
+      RepoToolsetBuildProj=$path
+      return
+    fi
+  fi
+  if [[ "$restore" != true ]]; then
+    ExitIfError 2 "Toolset version $RepoToolsetVersion has not been restored."
+  fi
 
   echo "Using repo tools version: $readjsonvalue"
```

Comment:
Shouldn't this line be just after the `ReadJson` call so it's printed on early-return too?